overops-functions/src/main/java/com/takipi/udf/deployment/DeploymentRoutingFunction.java

```diff
@@ -103,44 +102,37 @@ private static void buildDeploymentRoutingViews(ContextArgs args, DeploymentRout
 								ISODateTimeFormat.dateTimeParser().parseDateTime(dep1.last_seen)))
 				.collect(Collectors.toList());
 
-		// map deployments to viewIds of their corresponding views
-		Map<String, String> deploymentViewIdMap = Maps.newHashMapWithExpectedSize(maxViews);
-
-		for (SummarizedView view : views.values()) {
-			String deploymentNameFromView = (input.prefix != null) ?
-											(view.name.replace(cleanPrefix(input), "")) :
-											(view.name);
-
-			deploymentViewIdMap.put(deploymentNameFromView, view.id);
-		}
+		maxViews = Math.min(maxViews, sortedDeployments.size());
 
 		// remove existing views that exceed maxViews
-		for (int i = maxViews; i < sortedDeployments.size(); i++) {
-			SummarizedDeployment currentDeployment = sortedDeployments.get(i);
-			String currentDeploymentViewId = deploymentViewIdMap.get(currentDeployment.name);
+		for (SummarizedDeployment deployment : sortedDeployments.subList(maxViews, sortedDeployments.size())) {
+			SummarizedView deploymentView = views.get(toViewName(input, deployment.name));
 
-			if (currentDeploymentViewId != null) {
-				ViewUtil.removeView(apiClient, serviceId, currentDeploymentViewId);
+			if (deploymentView != null) {
+				System.out.println("Removing view " + deploymentView.id + " for deployment " + deployment.name);
+				ViewUtil.removeView(apiClient, serviceId, deploymentView.id);
 			}
 		}
 
 		// create missing views up to maxViews
 		List<ViewInfo> newDeploymentViewsInfo = Lists.newArrayList();
 
-		for (int i = 0; i < Math.min(sortedDeployments.size(), maxViews); i++) {
-			String currentDeploymentName = sortedDeployments.get(i).name;
-			String currentDeploymentViewId = deploymentViewIdMap.get(currentDeploymentName);
+		for (SummarizedDeployment deployment : sortedDeployments.subList(0, maxViews)) {
```

Comment:
Hmm...That doesn't seem right. We remove all deployment after the `maxViews` mark. And then add view up to the `maxViews` mark. doesn't that mean that latest deployments might never get added?

Reply:
The deployments are sorted above by descending last_seen so latest deployment is at index 0